### PR TITLE
[CI] Cherry pick azp related files from master branch

### PR DESCRIPTION
**Pull request by liushilongbuaa**
Signed-off-by: shilongliu <shilongliu@microsoft.com>

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012

### CODE REVIEW

**Comment by lguohan** [comment]
how is this validated?

**Comment by liushilongbuaa** [comment]
> how is this validated?

There is only cherry-pick and one commit to catch up with master branch. No other changes.
https://dev.azure.com/mssonic/build/_build/results?buildId=14377&view=results

**Comment by liushilongbuaa** [comment]
Other potential changes or bugs is not included, for easy review

File: .azure-pipelines/build-template.yml [hunk 1]
```diff
@@ -0,0 +1,125 @@
+parameters:
```
Comment (xumia) [inline .azure-pipelines/build-template.yml:1]: The file can be removed, it is no use, right?

**Review by xumia — APPROVED** [review]

**Pull request merged**